app/utils/timezone.js

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import DeviceInfo from 'react-native-device-info';
+
+export function getDeviceTimezone() {
+    return DeviceInfo.getTimezone();
+}
+
+export function isTimezoneEnabled(state) {
```

Comment:
Why not also check the server version here? We're still checking that in multiple places